src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -821,9 +821,10 @@ private void CheckModifiers(bool isExplicitInterfaceImplementation, Location loc
             {
                 diagnostics.Add(ErrorCode.ERR_AbstractNotVirtual, location, this.Kind.Localize(), this);
             }
-            else if (ContainingType.IsSealed && this.DeclaredAccessibility.HasProtected() && !this.IsOverride)
+            else if (ContainingType.IsSealed && this.DeclaredAccessibility.HasProtected() && !this.IsOverride &&
+                     ReportProtectedMemberInSealedTypeError(location, diagnostics))
```

Comment:
>ReportProtectedMemberInSealedTypeError [](start = 21, length = 38)

Consider keeping the diagnostic here, but just having a boolean property `ShouldReportProtectedMemberInSealedTypeError` (true for this type, false for derived type)